src/EventStore.Core/Caching/DynamicLRUCache.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Threading;
+using EventStore.Core.DataStructures;
+
+namespace EventStore.Core.Caching {
+	public class DynamicLRUCache<TKey, TValue> : LRUCache<TKey, TValue>, IDynamicLRUCache<TKey, TValue> {
+		const int ResizeBatchSize = 100000;
+
+		public DynamicLRUCache(long capacity, Func<TKey, TValue, int> calculateItemSize = null)
+			: base(capacity, calculateItemSize) { }
+
+		public void Resize(long newCapacity, out int removedCount, out long removedSize) {
```

Comment:
tbh i'd say this operation is invasive/coupled enough to the implementation of the LRUCache that it warrants putting it in the LRUCache itself, and get rid of this derived class (and then probably the dynamic interface too)

Reply:
fixed in: 890b0f3b3c73cc857bcd9868411d82cb01a38727